Feature/tabs anchor trigger

Clement-coder opened the pull request:
## Linked Issue

Closes #3472

## Description

Adds support for anchor (<a>) tags as tab triggers in TabsControl.
Now both <button> and <a> elements can be used to switch tabs.

## Checklist

Please read and apply all [contribution requirements](https://skeleton.dev/docs/resources/contribute/).

- [x] Your branch should be prefixed with: `docs/`, `feature/`, `chore/`, `bugfix/`
- [x] Contributions should target the `main` branch
- [ ] Documentation should be updated to describe all relevant changes
- [ ] Run `pnpm check` in the root of the monorepo
- [ ] Run `pnpm format` in the root of the monorepo
- [ ] Run `pnpm lint` in the root of the monorepo
- [ ] Run `pnpm test` in the root of the monorepo
- [ ] If you modify `/package` projects, please supply a Changeset

## Changsets

[View our documentation](https://skeleton.dev/docs/resources/contribute/get-started#changesets) to learn more about [Changesets](https://github.com/changesets/changesets). To create a Changeset:

1. Navigate to the root of the monorepo in your terminal
2. Run `pnpm changeset` and follow the prompts
3. Commit and push the changeset before flagging your PR review for review.

endigo9740 commented on packages/skeleton-svelte/src/components/Tabs/TabsControl.svelte:
```diff
@@ -3,6 +3,9 @@
 
 	const {
+		// New prop to choose element type
+		as = 'button', // 'button' or 'a'
```
No need for an explicit prop as this should be determined if an `href` is present. Zag likely provides this as part of the `ctx` or `state`. From their docs looks like there should be a [navigate() method](https://zagjs.com/components/react/tabs#machine-context) for this purpose.

endigo9740 commented on packages/skeleton-svelte/src/components/Tabs/TabsControl.svelte:
```diff
@@ -19,33 +22,59 @@
-</button>
+<!-- @component Individual tab trigger (anchor or button) -->
+
+{#if as === 'a'}
```
No need for duplication of code like this, as Svelte provides the `<svelte:element>` to solve for dynamic element types. See the docs here for details on the usage:
https://svelte.dev/docs/svelte/svelte-element

React offers a similar solution. Just note the use of capital letters for defining the dynamic element:
https://stackoverflow.com/a/33471928

endigo9740 commented on sites/skeleton.dev/src/lib/__tests__/Tabs.test.js:
```diff
@@ -0,0 +1,45 @@
+import { render, fireEvent } from '@testing-library/svelte';
```
Once again, please reference our existing components. We never separate our test cases like this. They are paired directly with the component in the component packages. Never on the doc app.

We cover this is the contribution docs:
https://www.skeleton.dev/docs/resources/contribute/components#add-components

And `Tabs.test.ts` already exists in the component directory:
https://github.com/skeletonlabs/skeleton/tree/main/packages/skeleton-svelte/src/components/Tabs

endigo9740 commented on sites/skeleton.dev/src/lib/components/Tabs.svelte:
```diff
@@ -0,0 +1,89 @@
+<script lang="ts">
```
This is left over from before and should be deleted.

endigo9740 commented on sites/skeleton.dev/src/routes/+page.svelte:
```diff
@@ -0,0 +1,17 @@
+<script>
```
Likewise, this file should be removed.

endigo9740 commented on sites/skeleton.dev/package.json:
```diff
@@ -12,10 +12,12 @@
 	},
 	"devDependencies": {
```
Please undo any changes to this and the `pnpm-lock.yaml` files, these changes nothing of value.

endigo9740 commented on playgrounds/skeleton-svelte/src/routes/components/tabs/+page.svelte:
```diff
@@ -139,4 +140,40 @@
+		<h2 class="h2">Anchor-Based Tabs Test</h2>
+		<Tabs value={group} onValueChange={(e) => (group = e.value)}>
+			<Tabs.Control>
+				<a
```
This doesn't appear to be valid code. The Tabs control represents the button/anchor. Why would multiple anchors be placed inside this?

endigo9740 reviewed (changes requested):
@Clement-coder this improved, but also still kind of a mess if I'm being honest. All your old changes have been resubmitted even though I stated they were incorrect yesterday.

The new changes are close, but should be improved with the specific notes added.

And you're still missing updates to the React component. It's fine if you wish to get the Svelte component sorted first, then React, but there was no indication you were doing this. I'm left to assume this was missed unintentionally.

I'll also go ahead and just ask straight up - are you using AI to implement these changes?

endigo9740 commented:
One more issue - please stop deleting the PR template. That information is important and it seems several of the requirements listed there were skipped or ignored. It's there both for your assistance, and for me as the maintainer to ensure all requirements requested were completed in full. I'll restore it this once.

Clement-coder commented:
Thanks for the honest feedback. I’m currently focused on getting the Svelte part right first, and then I’ll move on to the React updates. I am using AI to assist, but I’m not relying on it blindly  I’m carefully reviewing and shaping everything myself to make sure it aligns with the task. I’ll clean things up and address all your feedback properly.

Clement-coder commented:
Made updates to the TabsControl component to use dynamic elements properly without code duplication. Added matching tests inside the component directory. Also restored package files and cleaned up leftover code as per your feedback. Please review and let me know if there are any other changes needed.

nullpointerexceptionkek commented:
Hi @Clement-coder 

The current PR results in a deploy failure. Your code should not result in www.skeleton.dev preview deploy failure.

Do you still intend to work on this PR? Or else I will be taking over the issue.

endigo9740 commented:
> Your code should not result in [www.skeleton.dev](http://www.skeleton.dev/) preview deploy failure.

@nullpointerexceptionkek Actually this is a known issue on our end. We had a PR merged that any fork will contain as well. We'll need to solve it on the main branch and then subsequent feature branches can update to include the fix. Hit me up on Discord if you would like to discuss.

@Clement-coder given the issues remaining, I'm going to go ahead and make an executive call here and close this PR. While I appreciate your attempt to help, using AI tends to lead to these sorts of issues. I'd recommend against using this if you wish to contribute again in the future.

For now we'll let @nullpointerexceptionkek take over. I'll reassign the original ticket asap.

Clement-coder commented:
Thanks for the feedback. I understand your decision and I’ve learned a lot. I’ll take time to improve and avoid relying on AI too much. I hope to contribute again with better code. Thanks for the chance.
